conda/env/env.py

```diff
@@ -207,7 +219,400 @@ def add(self, package_name):
+
+        lines = [
+            f"name: {self.name if self.name else '<none>'}",
+            "options:",
+            textwrap.indent(
+                "\n".join([f"{key}: {value}" for key, value in self.options.items()]),
+                prefix="  ",
+            ),
+            "configuration:",
+            textwrap.indent(repr(self.config), "  "),
+            "requirements:",
+            textwrap.indent(repr(self.requirements), "  "),
+            "groups:",
+            textwrap.indent(
+                "\n".join(groups_lines),
+                "  ",
+            ),
+        ]
+        return "\n".join(lines)
+
+    def to_dict(self) -> dict:
+        """Return the environment as a dictionary.
+
+        :return: The environment as a dictionary
+        """
+        return self.serialize()
+
+    @classmethod
+    def from_history(cls, prefix: os.PathLike) -> EnvironmentV2:
+        """Create an Environment from the history of the requested packages in a prefix.
+
+        :param prefix: Prefix of the target conda environment
+        :return: EnvironmentV2 object containing all the user-requested conda packages
+        """
+        name = get_env_name(prefix)
+
+        requirements = []
+        for spec in History(prefix).get_requested_specs_map().values():
+            requirements.append(str(spec))
+
+        return EnvironmentV2.from_dict(
+            {
+                "name": name if name else None,
+                "requirements": requirements,
+                "variables": PrefixData(
+                    prefix, pip_interop_enabled=True
+                ).get_environment_env_vars(),
+            }
+        )
+
+    @classmethod
+    def from_prefix(cls, prefix: os.PathLike) -> EnvironmentV2:
+        """Create an Environment from the prefix of an environment on disk.
+
+        :param prefix: Prefix of the target conda environment
+        :return: EnvironmentV2 object containing all the conda packages in the prefix
+        """
+        pfd = PrefixData(prefix, pip_interop_enabled=True)
+        variables = pfd.get_environment_env_vars()
+
+        precs = tuple(PrefixGraph(pfd.iter_records()).graph)
+        grouped_precs = groupby(lambda x: x.package_type, precs)
+        conda_precs = sorted(
+            (
+                *grouped_precs.get(None, ()),
+                *grouped_precs.get(PackageType.NOARCH_GENERIC, ()),
+                *grouped_precs.get(PackageType.NOARCH_PYTHON, ()),
+            ),
+            key=lambda x: x.name,
+        )
+
+        pip_precs = sorted(
+            (
+                *grouped_precs.get(PackageType.VIRTUAL_PYTHON_WHEEL, ()),
+                *grouped_precs.get(PackageType.VIRTUAL_PYTHON_EGG_MANAGEABLE, ()),
+                *grouped_precs.get(PackageType.VIRTUAL_PYTHON_EGG_UNMANAGEABLE, ()),
+            ),
+            key=lambda x: x.name,
+        )
+
+        requirements = ["=".join((a.name, a.version)) for a in conda_precs]
```

Comment:
I think `PackageRecord` has a `to_match_spec()` method or similar. And also, why are we not adding the build string?

Reply:
This was actually ripped from `conda.env.nev.from_environment`, so that's why it may look familiar. I'll add in the build string now.